src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentValidatorUTest.java

```diff
@@ -0,0 +1,148 @@
+        attendanceSection.setCanvasSectionId(CANVAS_SECTION_ID);
+
+        attendanceAssignment = new AttendanceAssignment();
+        attendanceAssignment.setAssignmentId(ASSIGNMENT_ID);
+        attendanceAssignment.setAssignmentName(ASSIGNMENT_NAME);
+        attendanceAssignment.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment.setGradingOn(true);
+        attendanceAssignment.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment.setAbsentPoints(0.0);
+        attendanceAssignment.setExcusedPoints(0.0);
+        attendanceAssignment.setTardyPoints(0.0);
+        attendanceAssignment.setPresentPoints(100.0);
+        attendanceAssignment.setAttendanceSection(attendanceSection);
+        attendanceAssignment.setStatus(AttendanceAssignment.Status.UNKNOWN);
+
+
+        assignment = new Assignment();
+        assignment.setPointsPossible(ASSIGNMENT_POINTS);
+        assignmentOptional = Optional.of(assignment);
+
+    }
+
+    @Test
+    public void validateAttendanceAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
+        assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+        Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
+    }
+
+    @Test
+    public void validateCanvasAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
+
+        assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+        Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
+    }
+
+    @Test
+    public void validateAttendanceAssignmentEmptyOptionalIdValidationError() throws Exception {
+        assignmentOptional = Optional.empty();
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, Long.toString(CANVAS_ASSIGNMENT_ID))).thenReturn(assignmentOptional);
+
+        assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+    }
+
+
+    @Test
+    public void validateAttendanceAssignmentNullCanvasIDError() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)